### PR TITLE
CLI to skip polling for periodic information for a port in DomInfoUpdateTask thread

### PR DESCRIPTION
**Pull request by mihirpat1**
MSFT ADO - 26801913

#### What I did
Created a CLI to enable/disable periodic polling through DomInfoUpdateTask thread in XCVRD.
The CLI handler related code changes in xcvrd can be found at https://github.com/sonic-net/sonic-platform-daemons/pull/443

#### Why I did this
With https://github.com/sonic-net/sonic-platform-daemons/pull/443, the DomInfoUpdateTask thread in xcvrd now polls periodically for reading CDB firmware version of transceiver to update the TRANSCEIVER_FIRMWARE_INFO table.
However, transceivers cannot handle multiple CDB commands issued from different sources at the same time.
This scenario is likely to occur during CDB firmware upgrade (triggered through `sfputil firmware download/run/commit`) and while issuing `sfputil show fwversion PORT_NAME` CLI wherein 1 or more CDB commands are issued to the transceiver. At the same time, the DomInfoUpdateTask thread can issue a CDB command to read the firmware version as part of its periodic polling.
Hence, in order to ensure that only one process accesses the CDB command feature on transceivers at a time, the `config interface transceiver dom PORT_NAME enable/disable` CLI has been created to allow user to disable background polling of DomInfoUpdateTask thread for the particular port before upgrading the CDB firmware or before issuing `sfputil show fwversion PORT_NAME` CLI.

#### How I did it
With the execution of `config interface transceiver dom PORT_NAME enable/disable` CLI, the `dom_polling` field in PORT table of CONFIG_DB will be modified with the below value
1. Upon executing `config interface transceiver dom PORT_NAME enable` CLI, the `dom_polling` field will be set to `enabled`
2. Upon executing `config interface transceiver dom PORT_NAME disable` CLI, the `dom_polling` field will be set to `disabled`

#### How to verify it
Please refer to the test details attached to https://github.com/sonic-net/sonic-platform-daemons/pull/443

### CODE REVIEW

**Comment by prgeor** [comment]
@mihirpat1 can you explain "Why I did this" too in  the PR description

**Comment by mihirpat1** [comment]
> @mihirpat1 can you explain "Why I did this" too in the PR description

@prgeor I have added this section now.

File: config/main.py [hunk 1]
```diff
@@ -5173,6 +5175,40 @@ def reset(ctx, interface_name):
...
+            ctx.fail("'interface_name' is None!")
+
+    if interface_name_is_valid(config_db, interface_name) is False:
+        ctx.fail("Interface name is invalid. Please enter a valid interface name!!")
+
+    port_table_entry = config_db.get_entry("PORT", interface_name)
+    if not port_table_entry:
+        ctx.fail("Interface {} does not exist".format(interface_name))
+
+    # Reject config if port's corresponding subport is not part of DOM_CONFIG_SUPPORTED_SUBPORTS
```
Comment (prgeor) [inline config/main.py:5206]: @mihirpat1 may be a comment here will help we are handling non-breakout port when subport=0 and in breakout case, subport=1 (only the 1st subport)

**Review by prgeor — APPROVED** [review]

File: config/main.py (hunk 1 above)
Comment (mihirpat1) [reply config/main.py]: @prgeor I have modified the comment now.

**Review by prgeor — APPROVED** [review]

**Comment by mihirpat1** [comment]
@StormLiangMS @yxieca - Can you please help with cherry-pick to 202305 and 202311?
MSFT ADO - 26801913

**Comment by StormLiangMS** [comment]
hi @mihirpat1  could you run test with 202305 image with this PR?

**Pull request merged**